gossip/src/crds.rs

```diff
@@ -485,21 +485,30 @@ impl Crds {
             // all associated values.
             let origin = CrdsValueLabel::LegacyContactInfo(*pubkey);
             if let Some(origin) = self.table.get(&origin) {
-                if now < origin.local_timestamp.saturating_add(timeout) {
+                if origin
+                    .value
+                    .wallclock()
+                    .min(origin.local_timestamp)
+                    .saturating_add(timeout)
+                    > now
+                {
```

Comment:
post-merge comment: prolly worth wrapping this common logic in a helper

Reply:
I have more follow up changes in this area, and will likely revise this again.